subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/http/HttpServer.groovy

```diff
@@ -165,18 +165,25 @@ class HttpServer extends ServerWithExpectations implements HttpServerFixture {
         private final String path
         private final File srcFile
         private final boolean revalidate
+        private final int blockingSeconds
 
-        SendFileAction(String path, File srcFile, boolean revalidate) {
+        SendFileAction(String path, File srcFile, boolean revalidate, int blockingSeconds = 0) {
             super("return contents of $srcFile.name")
             this.srcFile = srcFile
             this.path = path
             this.revalidate = revalidate
+            this.blockingSeconds = blockingSeconds
         }
 
         void handle(HttpServletRequest request, HttpServletResponse response) {
             if (beforeHandle) {
                 beforeHandle.execute(request)
             }
+
+            if (blockingSeconds > 0) {
+                Thread.sleep(blockingSeconds * 1000)
```

Comment:
❌ We shouldn't introduce sleeps here. We have `CyclicBarrierHttpServer` for this kind of use case. Please use that instead of the plain `HttpServer`.